lib/dns/cache_test.go

```diff
@@ -215,8 +215,8 @@ func TestConcurrency(t *testing.T) {
 
 	wg.Wait()
 
-	// We need to have an empty cache now
-	if c.Count() != 0 {
-		t.Fatalf("Failed to remove eveything")
+	// We should remove at least one item from the cache
+	if c.Count() == samplesize {
```

Comment:
Don't understand this check. Shouldn't the old check succeed? This ones seems very ambiguous.